hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -239,16 +285,20 @@ def set_element(self, val, result, idx, *, fill=False):
             result[idx] = val
         except TypeError as err:
             raise InvalidArgument(
-                f"Could not add element={val!r} of {val.dtype!r} to array of "
-                f"{result.dtype!r} - possible mismatch of time units in dtypes?"
+                f"Could not add element={getattr(val, 'dtype', type(val))!r} of "
+                f"{getattr(val, 'dtype', type(val))!r} to array of "
+                f"{getattr(result, 'dtype', type(result))!r} - possible mismatch of time units in dtypes?"
```

Comment:
I think these expressions would be easier to read outside of the f-string.  It also looks like the dtype has been duplicated into the value position.  For the type, consider using repr for dtypes and get_pretty_function_representation for types.

Reply:
Sorry, that's `hypothesis.internal.reflection.get_pretty_function_description`